File: server/network/packet/CurrentVersionPacket.go
package packet

import (
	"gardon.local/server/util"
	"io"
)

type CurrentVersionPacket struct {
	Packet
	firmwareChecksum string
}

func (pkt *CurrentVersionPacket) ID() byte {
	return 0
}

func (pkt *CurrentVersionPacket) Write(buf io.Writer) error {
	return nil
}

func (pkt *CurrentVersionPacket) Read(buf io.Reader) error {
	readString, err := util.ReadString(buf)
	if err != nil {
		return err
	}

	pkt.firmwareChecksum = readString
	return nil
}

func (pkt *CurrentVersionPacket) FirmwareChecksum() string {
	return pkt.firmwareChecksum
}

File: client/src/networking/packet/CurrentVersionPacket.cpp
#include "CurrentVersionPacket.h"

namespace Network { 
    namespace Packet {
        uint8_t CurrentVersionPacket::id() {
            return 0;
        }

        size_t CurrentVersionPacket::estLength() {    
            return 2 + this->_firmwareChecksum.length();
        }

        void CurrentVersionPacket::read(Buffer& buffer) {

        }

        void CurrentVersionPacket::write(Buffer& buffer) {
            buffer.writeString(this->_firmwareChecksum);
        }

        String CurrentVersionPacket::firmwareChecksum() {
            return this->_firmwareChecksum;
        }

        void CurrentVersionPacket::firmwareChecksum(String firmwareChecksum) {
            this->_firmwareChecksum = firmwareChecksum;
        }
    }
}